s3backgrounddelete/s3backgrounddelete/object_recovery_scheduler.py

```diff
@@ -219,10 +219,33 @@ def schedule_periodically(self):
         def periodic_run(scheduler):
             """Add key value to queue using scheduler."""
             #Conditionally importing ObjectRecoveryRabbitMq/ObjectRecoveryMsgbusConsumer when config setting says so.
+            from s3backgrounddelete.object_recovery_msgbus import ObjectRecoveryMsgbus
+
+            if not self.producer:
+                self.producer = ObjectRecoveryMsgbus(
+                    self.config,
+                    self.logger)
+
+            maxKeys = self.config.get_max_keys()
+            count = self.producer.get_count()
+            threshold = 0.2*maxKeys
+            self.logger.info("threshold is : " + str(threshold))
+            self.logger.info("count of unread msgs is : " + str(count))
+
             if self.config.get_messaging_platform() == MESSAGE_BUS:
-                self.add_kv_to_msgbus()
+                if int(count) < threshold:
+                    self.logger.debug("count is less than 60 percent of threshold.")
+                    self.add_kv_to_msgbus()
+                else:
+                    #do nothing
+                    self.logger.debug("count is more than threshold")
```

Comment:
Chang log message to something like  "Queue has more than threshhold value. Hence skipping the add_kv_to_msgbus()"

Reply:
done.